mlx/fast.cpp

```diff
@@ -529,9 +529,12 @@ array scaled_dot_product_attention(
    * * CPU implementation
    * * batch size > 1
    * * query sequence length > 1
+   * * query sequence length < 16 (e.g. for speculative decoding)
+   * * query sequence length > 16 && non-null mask (causal attention)
```

Comment:
```suggestion
   * * query sequence length > 16 (causal attention)
```

`non-null mask` is mentioned below.